engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```
@@ -172,78 +232,211 @@ private void validate(SnapshotInfo snapshotInfo) {
+                    _volumeService.grantAccess(snapshotInfo, hostVO, srcDataStore);
+                }
+
+                Map<String, String> srcDetails = getSnapshotDetails(snapshotInfo);
+
+                copyCommand.setOptions(srcDetails);
+
+                copyCmdAnswer = (CopyCmdAnswer)_agentMgr.send(hostVO.getId(), copyCommand);
             }
             catch (Exception ex) {
-                s_logger.debug(ex.getMessage(), ex);
+                throw new CloudRuntimeException(ex.getMessage());
             }
+            finally {
+                try {
+                    _volumeService.revokeAccess(snapshotInfo, hostVO, srcDataStore);
+                }
+                catch (Exception ex) {
+                    s_logger.debug(ex.getMessage(), ex);
+                }
 
-            if (copyCmdAnswer == null || !copyCmdAnswer.getResult()) {
-                if (copyCmdAnswer != null && copyCmdAnswer.getDetails() != null && !copyCmdAnswer.getDetails().isEmpty()) {
-                    errMsg = copyCmdAnswer.getDetails();
+                if (copyCmdAnswer == null || !copyCmdAnswer.getResult()) {
+                    if (copyCmdAnswer != null && !StringUtils.isEmpty(copyCmdAnswer.getDetails())) {
+                        errMsg = copyCmdAnswer.getDetails();
+                    }
+                    else {
+                        errMsg = "Unable to perform host-side operation";
+                    }
                 }
-                else {
-                    errMsg = "Unable to perform host-side operation";
+
+                try {
+                    if (errMsg == null) {
+                        snapshotInfo.processEvent(Event.OperationSuccessed);
+                    }
+                    else {
+                        snapshotInfo.processEvent(Event.OperationFailed);
+                    }
+                }
+                catch (Exception ex) {
+                    s_logger.debug(ex.getMessage(), ex);
```

Why not log the ignored exception to ``WARN``?